### DISCUSSION TITLE
如何优化加载图片

### DISCUSSION THREAD

**Discussion by huangzhengyi** [netease-im/NIM_Duilib_Framework#397] [General]
如果直接在UI 线程SetBkImage ，如果图片加载的时候耗时较长，那就会导致UI短暂的卡顿
有什么更好的解决方案，比如先用一个单独的线程先将图片加载到cache，然后再setbkimage ?

**Comment by nmgwddj** [comment]
是什么场景需要展示多大的图片？

**Reply by huangzhengyi** [reply]
类似于预览图，需要实时更新
100个控件，显示不同的PNG 图片，每张大概25KB
图片已经通过其他线程异步下载到本地，现在就缺少优化setbkimage 的时候不卡顿

**Reply by huangzhengyi** [reply]
每10秒左右刷新一次，每次100张，每张大概25KB

**Comment by nmgwddj** [comment]
尝试将 100 个图片拼接，使用一个 image 控件显示。

**Reply by huangzhengyi** [reply]
每张预览图对应一台设备，控件后续还要选中点击操作的，合并一个之后无法知道点击的是谁了

**Reply by nmgwddj** [reply]
> 每张预览图对应一台设备，控件后续还要选中点击操作的，合并一个之后无法知道点击的是谁了

同时显示 100 张图片的方案一定是不可行的，没有什么 UI 的交互会这样来设计。
一种方案就是上面提到的，将图片拼接，按点击位置获取对应数据元素进行操作。
另外一种方案是翻页，每页只渲染部分图片。

**Reply by huangzhengyi** [reply]
我现在唯一能想到的，先通过非UI线程 GlobalManager::GetImage 将图片先加载到内存，然后让UI再setbkimage

**Reply by nmgwddj** [reply]
这样的方案也可以。只是我有一些疑虑，UI 展示 100 张图片是给人还是机器看？如果机器有分析能力为什么还需要展示出来？如果给人看这种明显不是良好的人机交互形式。
